### ISSUE TITLE
Error `wrong-type-argument hash-table-p nil` happens when I invoke elfeed

### ISSUE THREAD

**Issue by yang-ling** [skeeto/elfeed#209]
## Environment Information

* OS: Archlinux
* Emacs Version: 25.1.1, Spacemacs version: latest on develop branch.
* Elfeed Version: 20170412.1442 from https://melpa.org/packages/

## Reproduce Step

1. Open Emacs
2. Invoke `elfeed`
3. Error happens.

## Error Detail

Spacemacs has a function `debug-on-error`. I use that to get the following information.

```lisp
Debugger entered--Lisp error: (wrong-type-argument hash-table-p nil)
  gethash("http://www.quora.com/rss" nil)
  elfeed-db-get-feed("http://www.quora.com/rss")
  rmh-elfeed-org-export-feed(("http://www.quora.com/rss" quick quora qu "Quora"))
  -each((("http://www.quora.com/rss" quick quora qu "Quora") ("http://www.williamlong.info/rss.xml" quick blog moon tech "月光博客") ("http://blog.lilydjwg.me/feed" quick blog tech yi "依云") ("https://about.gitlab.com/atom.xml" quick blog tech gitlab gl "Gitlab") ("https://emacs-china.org/latest.rss" quick emacs tech ec "Emacs China") ("http://www.reddit.com/r/emacs/.rss" quick reddit emacs tech "Emacs") ("http://planet.emacsen.org/atom.xml" quick emacs pe tech "Planet Emacsen") ("http://blog.binchen.org/rss.xml" quick blog emacs chen tech "Chen's blog") ("http://www.daemonology.net/hn-daily/index.rss" quick news tech hn "HackerNews") ("http://www.chinagfw.org/feeds/posts/default" quick blog gfw tech "GFW") ("http://lwn.net/headlines/rss" quick tech lwn news "LWN") ("http://www.zhihu.com/rss" quick zhihu zh "知乎") ("http://feeds.feedburner.com/ruanyifeng" quick blog tech ruan "阮一峰") ("http://hongzhang0823.wix.com/ielts7/feed.xml" slow english study "雅思") ("http://eller86.hatenablog.jp/feed" slow blog tech toda "Toda") ("http://onehd.herokuapp.com/" lunchtime one han "一个 韩寒") ("http://comicfeeds.chrisbenard.net/view/dilbert/default" lunchtime dilbert pic di "Dilbert") ("http://aprogrammerslife.info/feed/" lunchtime proglife pic pl "Programmer's Life") ("http://xkcd.com/rss.xml" lunchtime xkcd pic "xkcd") ("http://feeds.feedburner.com/uclick/garfield" lunchtime cat pic "加菲猫") ("http://hdwo.net/feed" lunchtime movie hd "海盗湾中文网") ("http://twocoldbackup.blogspot.com/feeds/posts/default" lunchtime han blog "韩寒博客备份") ("http://www.reddit.com/r/funny/.rss" lunchtime reddit rf "Reddit Funny") ("http://www.feeddiy.com/rss/AJ3uQr" lunchtime pic girl mz jiandan "煎蛋妹子图") ("http://pocoo.tumblr.com/rss" lunchtime pic girl mz tumblr wakong "tumblr 妹子图") ("http://enmury.tumblr.com/rss" lunchtime pic girl mz tumblr iam "Imagine a man staring at a screen That’s you") ("http://www.lifehack.org/feed/" killtime lifehack lh "Lifehack") ("https://gitlab.com/RagefireChasm/sc.atom?private_token=GsTyRR2QbgbgmkMtwQeC" killtime gitlab boye "薄爷的项目的 Gitlab 追踪")) rmh-elfeed-org-export-feed)
  rmh-elfeed-org-process(("~/.emacs.d/private/my-elfeed-config/feed.org") "elfeed")
  ad-Advice-elfeed(#[0 "\301\302 !\210\303=?\205
```

## Note

* I have tried re-install Spacemacs and use its default `.spacemacs` config file and only add `Elfeed` layer. Same result.
* I have searched this error message on github, but no luck. Is it only me who has this error?

**Comment by skeeto** [comment]
This looks like a corrupt index file. If you take these steps:

1) Exit Emacs completely.
2) Move/delete the Elfeed "index" file (under elfeed-db-directory).
3) Restart Emacs.

Does the error persist?

As for the cause of the corruption, it may be related to having a
malformed elfeed-feeds. Is that your elfeed-feeds configuration in the
debug? I see a string at the end of each list, which isn't permitted.
Maybe it causes the database to only partially load, then later it's
saved (corrupting it), though I was unable to reproduce any of this
myself.

I pushed a commit tagged with this issue, but I don't actually expect it
to affect anything. It was a potential future bug that would manifest
exactly like this issue, and I noticed it while thinking about the
current problem.
